app/src/test/java/org/cru/godtools/databinding/ToolDetailsFragmentBindingTest.kt

```diff
@@ -0,0 +1,45 @@
+package org.cru.godtools.databinding
+
+import android.app.Application
+import android.view.LayoutInflater
+import android.view.View
+import androidx.lifecycle.MutableLiveData
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.cru.godtools.xml.model.Manifest
+import org.cru.godtools.ui.tooldetails.ToolDetailsActivity
+import org.cru.godtools.ui.tooldetails.ToolDetailsFragment
+import org.cru.godtools.xml.util.getXmlParserForResource
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.Assert.assertEquals
+import org.robolectric.Robolectric
+import org.robolectric.annotation.Config
+import java.util.Locale
+
+private const val TOOL_CODE = "test"
+
+@RunWith(AndroidJUnit4::class)
+@Config(application = Application::class)
+class ToolDetailsFragmentBindingTest {
+    private lateinit var binding: ToolDetailsFragmentBinding
+
+    @Before
+    fun createBinding(){
+        val activityController = Robolectric.buildActivity(ToolDetailsActivity::class.java)
+        binding = ToolDetailsFragmentBinding.inflate(LayoutInflater.from(activityController.get()), null, false)
+    }
+
+    @Test
+    fun verifyWithTips() {
+        val manifest = Manifest(TOOL_CODE, Locale.ENGLISH, getXmlParserForResource("manifest_tips_invalid.xml")){
+            getXmlParserForResource(it)
+        }
```

Comment:
instead of parsing xml to get the manifest, just create it using something like this:
```kotlin
val manifest = Manifest(tips = { listOf(Tip(it, "tip1")) })
```

Reply:
@frett sorry I didn't see this I used mock values.  let me know if you are OK with that approach

Reply:
use these testing constructors, it easier than having to mock everything for tool models